### PR TITLE
feat: add hook build:config

### PR DESCRIPTION
**Pull request by clarkdo**
## Types of changes

- [ ] Bug fix (a non-breaking change which fixes an issue)
- [ ] New feature (a non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Description

I didn't add any param for now since we already have `extend` for customising webpack config and this hook should avoid introducing more side effects.

## Checklist:

- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly. (PR: #)
- [ ] I have added tests to cover my changes (if not applicable, please state why)
- [ ] All new and existing tests are passing.

### CODE REVIEW

**Comment by manniL** [comment]
I'm wondering whether we should be more explicit (e.g. `webpackConfig`) and use the new patter (`builder:`)?

Also, we need docs (as usual) explaining the difference between the different methods of adding stuff to the webpack config (e.g. via extends).

File: packages/webpack/src/builder.js [hunk 1]
```diff
@@ -57,6 +57,12 @@ export class WebpackBundler {
       webpackConfigs.push(this.getWebpackConfig('Server'))
     }
 
+    const additionalConfigs = await this.buildContext.nuxt.callHook('build:config')
+
+    if (additionalConfigs) {
+      webpackConfigs.push(...[].concat(additionalConfigs))
```
Comment (pimlie) [inline packages/webpack/src/builder.js]: Why the `[].concat`? If the array contains objects then after the concat its still an array to the same references right?

Comment (clarkdo) [reply packages/webpack/src/builder.js]: The concat here is for supporting return value either array or object as single or multiple builds

**Comment by clarkdo** [comment]
@manniL Sorry, I don’t know the new hook pattern decision, is the new pattern corresponding to package? So maybe use hook name as ‘webpack:config’ ?

File: packages/webpack/src/builder.js (hunk 1 above)
Comment (pimlie) [reply packages/webpack/src/builder.js]: Thanks, actually I didnt know you could return values with [`callHook`](https://github.com/jsless/hable/blob/master/src/hable.js#L45). Maybe I am just overseeing something simple but there is no return statement or whatever, so how does it even return a value?

**Comment by clarkdo** [comment]
Will open another pr

File: packages/webpack/src/builder.js (hunk 1 above)
Comment (clarkdo) [reply packages/webpack/src/builder.js]: Yes, that's true, it has been changed for a while, I've changed in new pr

**Comment by manniL** [comment]
> y, I don’t know the new hook pattern decision, is the new pattern corresponding to package? So maybe use hook name as ‘webpack:config’ ?

https://github.com/nuxt/nuxt.js/pull/6329#issuecomment-526927022

**Pull request merged**